src/Fixer/Operator/OperatorLinebreakFixer.php

```diff
@@ -0,0 +1,262 @@
+{
+    /**
+     * @internal
+     */
+    const BOOLEAN_OPERATORS = [
+        '&&' => true,
+        '||' => true,
+        'and' => true,
+        'or' => true,
+        'xor' => true,
+    ];
+
+    /**
+     * @internal
+     */
+    const NON_BOOLEAN_OPERATORS = [
+        '=' => true,
+        '*' => true,
+        '/' => true,
+        '%' => true,
+        '<' => true,
+        '>' => true,
+        '|' => true,
+        '^' => true,
+        '+' => true,
+        '-' => true,
+        '&' => true,
+        '&=' => true,
+        '.=' => true,
+        '/=' => true,
+        '=>' => true,
+        '==' => true,
+        '>=' => true,
+        '===' => true,
+        '!=' => true,
+        '<>' => true,
+        '!==' => true,
+        '<=' => true,
+        '-=' => true,
+        '%=' => true,
+        '*=' => true,
+        '|=' => true,
+        '+=' => true,
+        '<<' => true,
+        '<<=' => true,
+        '>>' => true,
+        '>>=' => true,
+        '^=' => true,
+        '**' => true,
+        '**=' => true,
+        '<=>' => true,
+        '??' => true,
+        '?' => true,
+        ':' => true,
+    ];
+
+    /** @var null|string[] */
+    private $operators;
+
+    /** @var bool */
+    private $onlyBooleans = false;
+
+    /** @var string */
+    private $position = 'beginning';
+
+    /**
+     * @return FixerDefinition
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Operators - when multiline - must always be at the beginning or at the end of the line.',
+            [new CodeSample('<?php
+function foo() {
+    return $bar ||
+        $baz;
+}
+')]
+        );
+    }
+
+    /**
+     * @return FixerConfigurationResolver
+     */
+    public function getConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('only_booleans', 'whether to limit operators to only boolean ones'))
+                ->setDefault($this->onlyBooleans)
+                ->setAllowedTypes(['bool'])
+                ->getOption(),
+            (new FixerOptionBuilder('position', 'whether to place operators at the beginning or at the end of the line'))
+                ->setDefault($this->position)
+                ->setAllowedValues(['beginning', 'end'])
+                ->getOption(),
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        parent::configure($configuration);
+
+        if (isset($configuration['only_booleans'])) {
+            $this->onlyBooleans = $configuration['only_booleans'];
+        }
+        if (isset($configuration['position'])) {
+            $this->position = $configuration['position'];
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return true;
```

Comment:
if `only_booleans` is configured I think a token-of-type-of-interest-is-found might be justified

Reply:
Done